src/org/thoughtcrime/securesms/MyLayoutManager.java

```diff
@@ -0,0 +1,49 @@
+package org.thoughtcrime.securesms;
+
+import android.content.Context;
+import android.os.Parcelable;
+
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import org.thoughtcrime.securesms.util.ViewUtil;
+
+/**
+ * Like LinearLayoutManager but you can set a starting position
+ */
+class MyLayoutManager extends LinearLayoutManager {
```

Comment:
can we have a more describing name for that?

also, overwriting a central function - why is that needed? this may cause troubles on different implementations.

i just had a look - signal uses SmoothScrollingLinearLayoutManager - is that comparable?

Reply:
No, I guess that Signal uses SmoothScrollingLinearLayoutManager in order to be able to set the speed of smooth scrolling. We need a custom LayoutManager in order to be able to set the starting position ( 5c5e267 ). I don't know how Signal does this but their code has diverged a lot from ours in this area, so that I decided it's easier to implement this myself (would have been at least some 100 lines to pull in)

I thought about renaming, too, but didn't know a good name. SetStartingPositionLinearLayoutManager?

Reply:
thanks for explaining. and yes, `SetStartingPositionLayoutManager` looks good to me.  maybe a bit long, however, it is not used that often and make things much clearer.